frontend/components/search-results.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { Link } from "react-router-dom";
+import PropTypes from 'prop-types';
+import { BadgeExamples } from './badge-examples';
+import badgeExampleData from '../../badge-examples.json';
+import { prepareExamples, predicateFromQuery } from '../lib/prepare-examples';
+import { baseUri, longCache } from '../constants';
+
+
+export default class SearchResults extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.props = props;
+  }
+
+  static propTypes = {
+    category: PropTypes.string,
+    query: PropTypes.string,
+    clickHandler: PropTypes.func.isRequired,
+  }
+
+  prepareExamples(category) {
+    const examples = category ? badgeExampleData.filter(example => example.category.id === category) : badgeExampleData;
+    return prepareExamples(examples, () => predicateFromQuery(this.props.query));
+  }
+
+  renderExamples() {
+    return (
+      <BadgeExamples
+        categories={this.preparedExamples}
+        onClick={this.props.clickHandler}
+        baseUri={baseUri}
+        longCache={longCache} />
+    );
+  }
+
+  renderCategories() {
+    return this.preparedExamples.map(function(category, i) {
+      return (
+        <Link to={'/examples/' + category.category.id} key={i}>
```

Comment:
I've learned from a linter rule on another project that `key` is not supposed to be an array index, rather something that persistently identifies a piece of content. The purpose of `key` is so React can be smart about adding and removing DOM elements. If some of the child items go away, their keys should stay the same. Could we use the category id instead?